Documentation/flexvolume.md

```diff
@@ -5,34 +5,64 @@ indent: true
+In these environments, the kubelet needs to be told to use a different flexvolume plugin directory that is accessible and writeable (`rw`).
 These steps need to be carried out on **all nodes** in your cluster.
 
-Please refer to all below sections that are applicable for you, they contain more information on your specific platform and/or Kubernetes version used.
+Please refer to the section that is applicable to your environment, it contains more information on flexvolume on your platform.
 
-```bash
---volume-plugin-dir=/var/lib/kubelet/volumeplugins
-```
-Restart Kubelet in order for this change to take effect.
+## Platform specific flexvolume path
+### For Atomic
+To find out which flexvolume directory path you need to set on the rook-operator in the , please look at the OpenShift docs of your version, [latest OpenShift Flexvolume docs](https://docs.openshift.org/latest/install_config/persistent_storage/persistent_storage_flex_volume.html#flex-volume-installation) (they also contain the flexvolume path for Atomic hosts).
+
+Continue with [Configuring the flexvolume path](#configuring-the-flexvolume-path) to configure Rook to use the flexvolume path.
 
-## For Kubernetes >= 1.9.x
-In Kubernetes >= `1.9.x`, you must provide the above set Flexvolume plugin directory when deploying the [rook-operator](/cluster/examples/kubernetes/ceph/operator.yaml) by setting the environment variable `FLEXVOLUME_DIR_PATH`. For example:
+### For ContainerLinux
+Use the [Most common writeable flexvolume path](#most-common-writeable-flexvolume-path) for the next steps.
+
+Continue with [Configuring the flexvolume path](#configuring-the-flexvolume-path) to configure Rook to use the flexvolume path.
+
+### For OpenShift
+See the [For Atomic](#atomic) section.
+
+### For Rancher
+Rancher provides an easy way to configure kubelet. This flag can be provided to the kubelet configuration template at deployment time or by using the `up to date` feature if Kubernetes is already deployed.
+
+Continue with [Configuring the flexvolume path](#configuring-the-flexvolume-path) to configure Rook to use the flexvolume path.
+
+### For Tectonic
+Follow [these instructions](tectonic.md) to configure the Flexvolume plugin for Rook on Tectonic during ContainerLinux node ignition file provisioning.
+
+Continue with [Configuring the flexvolume path](#configuring-the-flexvolume-path) to configure Rook to use the flexvolume path.
+
+### For custom containerized kubelet
+Use the [Most common writeable flexvolume path](#most-common-writeable-flexvolume-path) for the next steps.
+
+If your kubelet is running as a (Docker, rkt, etc) container you need to make sure that this directory from the host is reachable by the kubelet inside the container.
+
+Continue with [Configuring the flexvolume path](#configuring-the-flexvolume-path) to configure Rook to use the flexvolume path.
+
+## Configuring the flexvolume path
```

Comment:
we are no longer mentioning the Kubelet's `--volume-plugin-dir` flag anywhere in this doc, isn't that really important if you need to tell the Kubelet to look somewhere besides its default location?

Reply:
I have added a section below so it is in order of:
1. Find where flexvolume path (and kubelet config "file") is located
2. Configure the Rook operator
3. Configure the kubelet